management/queue.go

```diff
@@ -581,3 +582,50 @@ func (m *queueManager) CompleteJobs(ctx context.Context, f StatusFilter) error {
 
 	return nil
 }
+
+func (m *queueManager) CompleteJobsByPrefix(ctx context.Context, f StatusFilter, prefix string) error {
+	if err := f.Validate(); err != nil {
+		return errors.WithStack(err)
+	}
+	if f == Completed {
+		return errors.New("invalid specification of completed job type")
+	}
+
+	for stat := range m.queue.JobStats(ctx) {
+		if stat.Completed {
+			continue
+		}
+		if !strings.HasPrefix(stat.ID, prefix) {
+			continue
+		}
+
+		switch f {
+		case Stale:
+			if stat.InProgress && time.Since(stat.ModificationTime) > m.queue.Info().LockTimeout {
```

Comment:
I read this to pick out jobs that _are_ stale, whereas I read the `InProgress` case to pick out jobs that are _not_ in progress.

Reply:
I am actually confused now, I copied and pasted this... but looking at all the management functions they each do different things with the `Stale` case.